fe/fe-core/src/main/java/com/starrocks/sql/analyzer/InsertAnalyzer.java

```diff
@@ -286,7 +263,102 @@ public static void analyze(InsertStmt insertStmt, ConnectContext session) {
+            table = getOLAPExternalTableMeta(database, (ExternalOlapTable) table);
+        }
+        return table;
+    }
+
+    private static Table makeTableFunctionTable(InsertStmt insertStmt) {
+        // fetch schema from query
+        QueryRelation query = insertStmt.getQueryStatement().getQueryRelation();
+        List<Field> allFields = query.getRelationFields().getAllFields();
+        List<Column> columns = allFields.stream().filter(Field::isVisible).map(field -> new Column(field.getName(),
+                field.getType(), field.isNullable())).collect(Collectors.toList());
+
+        // parse table function properties
+        Map<String, String> props = insertStmt.getTableFunctionProperties();
+        boolean writeSingleFile = Boolean.parseBoolean(props.get("single"));
+        String path = props.get("path");
+        String format = props.get("format");
+        String partitionBy = props.get("partition_by");
+
+        // validate properties
+        if (path == null) {
+            throw new SemanticException(
+                    "path is mandatory in table function. \"path\" = \"hdfs://path/to/your/location/prefix\"");
+        }
+
+        if (format == null) {
+            throw new SemanticException("format is mandatory in table function. " +
+                    "Use \"path\" = \"parquet\" as only parquet format is supported now");
+        }
+
+        if (!format.equalsIgnoreCase("parquet")) {
+            throw new SemanticException("use \"path\" = \"parquet\", as only parquet format is supported now");
+        }
+
+        if (writeSingleFile && partitionBy != null) {
+            throw new SemanticException("cannot use partition by and single simultaneously");
+        }
+
+        if (writeSingleFile) {
+            return new TableFunctionTable(path, format, columns, null, true);
+        }
+
+        if (partitionBy == null) {
+            // prepend `data_` if path ends with forward slash
+            if (path.endsWith("/")) {
+                path += "data_";
+            }
+            return new TableFunctionTable(path, format, columns, null, false);
+        }
+
+        // extra validation for using partitionBy
+        if (!path.endsWith("/")) {
+            throw new SemanticException(
+                    "To enable partition_by, path should be a directory ends with forward slash(/)");
+        }
+
+        List<String> columnNames = columns.stream().map(Column::getName).collect(Collectors.toList());
+
+        // parse and validate partition columns
+        List<String> partitionColumnNames = Arrays.asList(partitionBy.split(","));
+        partitionColumnNames.replaceAll(String::trim);
+        partitionColumnNames = partitionColumnNames.stream().distinct().collect(Collectors.toList());
+
+        List<String> unmatchedPartitionColumnNames = partitionColumnNames.stream().filter(col ->
+                !columnNames.contains(col)).collect(Collectors.toList());
```

Comment:
```suggestion
                List<String> unmatchedPartitionColumnNames = partitionColumnNames.stream()
                .filter(col -> !columnNames.contains(col))
                .collect(Collectors.toList());
```
I found it more comfortable to break the line like this.

Reply:
agree!